lib/application.js

```diff
@@ -116,9 +117,13 @@ module.exports = class Application extends Emitter {
   use(fn) {
     if (typeof fn !== 'function') throw new TypeError('middleware must be a function!');
     if (isGeneratorFunction(fn)) {
-      deprecate('Support for generators will be removed in v3. ' +
-                'See the documentation for examples of how to convert old middleware ' +
-                'https://github.com/koajs/koa/blob/master/docs/migration.md');
+      process.emitWarning([
```

Comment:
https://nodejs.org/api/process.html#process_avoiding_duplicate_warnings

Reply:
I've seen that. But this refers to _the same_ warning and this warning will be emitted once per middleware (and I've added middleware name of first line of code for convenience). So, it's completely in line with Node documentation suggestion - _the same_ warning emits just once per process.

Reply:
Many times, you may have a graceful restart in the app where the Koa app is in a factory, and the same app may be constructed multiple times in the same process. This would emit a warning for each use for each time the same app is constructed. Perhaps then if the idea is to emit once per name, then track which names have already been emitted instead.

Reply:
Isn't that too much functionality for deprecated feature? I can do it for sure, but don't want to have another 500+ lines of code :-)

Reply:
Then why not just emit it only once? You said the idea is to emit once per generator, but I showed that that is not the case. What exactly is the case, then? This seems to emit unconditionally forever, against the Node.js guidelines for usage??

Reply:
The original deprecations message will also appear several time in cases you suggesting (creating several Koa instances in the same process). So, we may be not at the _best practices_ of Node in that edge case, but definitely is no way more noise than with external dependency.

Reply:
> The original deprecations message will also appear several times in cases you suggesting (creating several Koa instances in the same process).

No it doesn't. Can you provide code that demonstrates otherwise?

Reply:
My bad, I've actually didn't set `DeprecationWarning` type, fixing now